flake.nix

```diff
@@ -23,96 +19,133 @@
+            };
           };
-        };
+        });
 
       };
 
       # Provide some binary packages for selected system types.
       packages = forAllSystems (system:
+        let
+          pkgSet = nixpkgsFor.${system};
+        in
         {
-          inherit (nixpkgsFor.${system}) hello;
+
+          inherit (pkgSet)
+            thunderbird thunderbird-with-extensions;
+
+          inherit (pkgSet.thunderbird-extensions)
+            enigmail
+            tbsync dav-4-tbsync eas-4-tbsync
+            autocrypt;
+
+          sample-thunderbird = pkgSet.thunderbird-with-extensions.override {
+            thunderbirdExtensions = with pkgSet.thunderbird-extensions; [
+              enigmail
+
+              tbsync
+              dav-4-tbsync
+              eas-4-tbsync
+            ];
+          };
+
         });
 
       # The default package for 'nix build'. This makes sense if the
       # flake provides only one package or there is a clear "main"
       # package.
-      defaultPackage = forAllSystems (system: self.packages.${system}.hello);
+      defaultPackage = forAllSystems (system: self.packages.${system}.thunderbird-with-extensions);
 
       # A NixOS module, if applicable (e.g. if the package provides a system service).
-      nixosModules.hello =
+      nixosModules.thunderbird =
         { pkgs, ... }:
         {
-          nixpkgs.overlays = [ self.overlay ];
+          imports =
+            [
+              ./modules/thunderbird.nix
+            ];
 
-          environment.systemPackages = [ pkgs.hello ];
-
-          #systemd.services = { ... };
+          nixpkgs.overlays = [ self.overlay ];
         };
 
       # Tests run by 'nix flake check' and by Hydra.
       checks = forAllSystems (system: {
-        inherit (self.packages.${system}) hello;
+        inherit (self.packages.${system})
+          thunderbird thunderbird-with-extensions
+          enigmail tbsync dav-4-tbsync eas-4-tbsync
+          sample-thunderbird;
 
         # Additional tests, if applicable.
-        test =
+        thunderbird-default =
           with nixpkgsFor.${system};
+          with self.packages.${system};
           stdenv.mkDerivation {
-            name = "hello-test-${version}";
-
-            buildInputs = [ hello ];
-
-            unpackPhase = "true";
-
+            name = "thunderbird-default-test";
+            unpackPhase = ":";
+            buildInputs = [ thunderbird ];
             buildPhase = ''
-              echo 'running some integration tests'
-              [[ $(hello) = 'Hello, world!' ]]
+              export HOME=$(pwd)
+              mkdir -p $HOME/fakeprofile
+              thunderbird --headless --profile $HOME/fakeprofile &
+              sleep 128
+              ls $HOME/fakeprofile/extensions | grep -q ${tbsync.emid} && exit 1 || echo "valid extensions"
+              [ "$(ls $HOME/fakeprofile/extensions | wc -l)" = "1" ] && echo "valid number of extensions" || exit 1
+            '';
```

Comment:
I think I'm missing something here: Why would you expect the tbsync extension if the "default" thunderbird?

Reply:
Thunderbird includes an extension by default (Lighting - Integrated Calendar).